source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -362,6 +362,11 @@ void ClientSslSocketFactory::onAddOrUpdateSecret() {
   stats_.ssl_context_update_by_sds_.inc();
 }
 
+bool ClientSslSocketFactory::isReady() const {
+  return config_->checkTlsCertificateEntityExists() &&
```

Comment:
I think you can simply make those two function into `isSecretsReady` in ContextConfig interface.

Reply:
done